tests/ga/test_extension.py

```diff
@@ -1351,18 +1351,12 @@ def test__old_handler_reports_failure_on_disable_fail_on_update(self, patch_get_
             exthandlers_handler.run()  # Download the new update the first time, and then we patch the download method.
             self.assertEqual(1, patch_get_disable_command.call_count)
 
-            new_version_args, new_version_kwargs = patch_report_event.call_args
-            self.assertFalse(new_version_kwargs['log_event'],
-                             "The final call to report_events should not log event")
-            self.assertEqual(new_version_args[0].ext_handler.properties.version, new_version,
-                             "The final report event call should be from the new version ext-handler "
-                             "as we want to report failure for the new version in the status blob")
-
-            old_version_args, old_version_kwargs = patch_report_event.call_args_list[1]
+            old_version_args, old_version_kwargs = patch_report_event.call_args
```

Comment:
Can we also assert that the `report_event` was only called once?

Reply:
Its actually called twice - 
- Once to report that download of the new version was successful
- Second time to report the disable failure

I think asserting that the last call to report_event was done to report the disable failure is good enough as there should be no other calls after this in this run.

Reply:
`patch_report_event.call_args` 
The `call_args` property of a patched object always picks up the last call parameters, so this is ensuring that we're only checking the last call

Reply:
In that case, can we explicitly assert the call args for both calls :)
You can use `call_args_list` to get all calls and their args.

Reply:
Good point, I'll send out an update. Thanks :)